source/adapters/hip/CMakeLists.txt

```diff
@@ -10,13 +10,16 @@ set(UR_HIP_PLATFORM "AMD" CACHE STRING "UR HIP platform, AMD or NVIDIA")
 
 # Set default ROCm installation directory
 set(UR_HIP_ROCM_DIR "/opt/rocm" CACHE STRING "ROCm installation dir")
-
-set(UR_HIP_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/include")
-
-set(UR_HIP_HSA_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/hsa/include")
-
-# Set HIP lib dir
-set(UR_HIP_LIB_DIR "${UR_HIP_ROCM_DIR}/lib")
+# Allow custom location of HIP/HSA include and HIP library directories
+if("${UR_HIP_INCLUDE_DIR}" STREQUAL "")
+  set(UR_HIP_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/include")
+endif()
+if("${UR_HIP_HSA_INCLUDE_DIR}" STREQUAL "")
+  set(UR_HIP_HSA_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/hsa/include")
+endif()
+if("${UR_HIP_LIB_DIR}" STREQUAL "")
+  set(UR_HIP_LIB_DIR "${UR_HIP_ROCM_DIR}/lib")
+endif()
```

Comment:
Using the cache for this would be fewer lines and has the benefit of also adding docs per option:

```cmake
# Allow custom location of HIP/HSA include and HIP library directories
set(UR_HIP_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/include" CACHE PATH "Path to the HIP include directory.")
set(UR_HIP_HSA_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/hsa/include" CACHE PATH "Path to the HSA include directory.")
set(UR_HIP_LIB_DIR "${UR_HIP_ROCM_DIR}/lib" CACHE PATH "Path to the HIP library directory.")
```

Then usage would just add `FORCE` to the end when setting a custom value

```cmake
# Allow custom location of HIP/HSA include and HIP library directories
set(UR_HIP_INCLUDE_DIR "/my/custom/hip/include" CACHE PATH "Path to the HIP include directory." FORCE)
set(UR_HIP_HSA_INCLUDE_DIR "/my/custom/hsa/include" CACHE PATH "Path to the HSA include directory." FORCE)
set(UR_HIP_LIB_DIR "/my/custom/hip/lib" CACHE PATH "Path to the HIP library directory." FORCE)
```

Reply:
My goal with these suggestions is to have all documented user configurable options/variables exist in the CMake cache in the same way `option()` would do.

I think then, the variables being set by internal logic should be separate that those used for user facing configuration varaibles.